src/main/kotlin/no/nav/familie/ba/sak/behandling/vedtak/vedtaksperiode/VedtaksperiodeService.kt

```diff
@@ -0,0 +1,54 @@
+package no.nav.familie.ba.sak.behandling.vedtak.vedtaksperiode
+
+import no.nav.familie.ba.sak.behandling.Behandlingutils
+import no.nav.familie.ba.sak.behandling.domene.Behandling
+import no.nav.familie.ba.sak.behandling.domene.BehandlingRepository
+import no.nav.familie.ba.sak.behandling.grunnlag.personopplysninger.PersongrunnlagService
+import no.nav.familie.ba.sak.behandling.grunnlag.personopplysninger.PersonopplysningGrunnlag
+import no.nav.familie.ba.sak.beregning.domene.AndelTilkjentYtelse
+import no.nav.familie.ba.sak.beregning.domene.AndelTilkjentYtelseRepository
+import no.nav.familie.ba.sak.beregning.domene.TilkjentYtelseRepository
+import org.springframework.stereotype.Service
+
+@Service
+class VedtaksperiodeService(
+        private val behandlingRepository: BehandlingRepository,
+        private val persongrunnlagService: PersongrunnlagService,
+        private val andelTilkjentYtelseRepository: AndelTilkjentYtelseRepository,
+        private val tilkjentYtelseRepository: TilkjentYtelseRepository
+) {
+
+    fun hentVedtaksperioder(behandling: Behandling): List<Vedtaksperiode> {
+        val iverksatteBehandlinger = Behandlingutils.hentIverksatteBehandlinger(behandlingRepository.finnBehandlinger(behandling.id), tilkjentYtelseRepository)
+
+        val forrigeIverksatteBehandling: Behandling? = Behandlingutils.hentForrigeIverksatteBehandling(
+                iverksatteBehandlinger = iverksatteBehandlinger,
+                behandlingFørFølgende = behandling
+        )
+
+        val forrigePersonopplysningGrunnlag: PersonopplysningGrunnlag? =
+                if (forrigeIverksatteBehandling != null) persongrunnlagService.hentAktiv(behandlingId = forrigeIverksatteBehandling.id) else null
+        val forrigeAndelerTilkjentYtelse: List<AndelTilkjentYtelse> =
+                if (forrigeIverksatteBehandling != null) andelTilkjentYtelseRepository.finnAndelerTilkjentYtelseForBehandling(
+                        behandlingId = forrigeIverksatteBehandling.id) else emptyList()
+
+        val personopplysningGrunnlag = persongrunnlagService.hentAktiv(behandlingId = behandling.id)
+                                       ?: return emptyList()
+        val andelerTilkjentYtelse =
+                andelTilkjentYtelseRepository.finnAndelerTilkjentYtelseForBehandling(behandlingId = behandling.id)
+
+        val utbetalingsperioder = mapTilUtbetalingsperioder(
+                andelerTilkjentYtelse = andelerTilkjentYtelse,
+                personopplysningGrunnlag = personopplysningGrunnlag
+        )
+
+        val opphørsperioder = mapTilOpphørsperioder(
+                forrigePersonopplysningGrunnlag = forrigePersonopplysningGrunnlag,
+                forrigeAndelerTilkjentYtelse = forrigeAndelerTilkjentYtelse,
+                personopplysningGrunnlag = personopplysningGrunnlag,
+                andelerTilkjentYtelse = andelerTilkjentYtelse
+        )
+
+        return listOf(utbetalingsperioder, opphørsperioder).flatten()
```

Comment:
Jeg har brukt denne måten for å slå sammen lister, men i går fant jeg ut at man bare kan plusse. Funker dette? :D 
```suggestion
        return utbetalingsperioder + opphørsperioder
```

Reply:
Jepp - kjempefint! Takk

Reply:
Kult @ninakjek 👍